CIPs/0017.md

```diff
@@ -0,0 +1,101 @@
+
+# CIP [0017]: Integrate Plumo Proofs into the lightest sync protocol
+
+- Date: 2020-07-27
+- Author: @lucasege
+- Status: WIP
+
+## Overview
+
+In order to continue supporting the dream of a seamless mobile sync experience, the lightest sync protocol must be extended to support PLUMO Zk-SNARKs which can quickly verify transitions from epoch X to epoch X + Y, where Y could be on the order of 10s-100s of epochs.
+
+The prover/verifier for plumo proofs have already been implemented, and the exposed go interface is available [here](https://github.com/celo-org/celo-bls-go)
+
+## Goals
+
+- Support adding/storing proofs on full nodes
+- Gossip these proofs to propogate them through the network
+- Extend the `les` protocol such that the server will provide these proofs in place of headers (when appropriate), and the client will consume/verify proofs accordingly.
+
+## Proposed Solution
+
+We propose to add two new RPCs: 
+* `AddProof(proof []byte, firstEpoch uint, lastEpoch uint, versionNumber uint)`
+* `Proofs() ([][]byte)`.
+
+`versionNumber` here refers to the snark circuit version, for more info refer to #upgradeability below.
+
+When an eventual "proving service" generates a new proof it will publish this proof to a full node that it is running, which will then propagate the proof throughout the network.
+
+This `AddProof` function will then utilize the `verificationKey` corresponding to the `versionNumber` and the corresponding validator sets for these epochs to verify the proof. If successful, it will then store the proof in a new database `plumoproofdata`, keyed by it's combination of first and last epoch. 
```

Comment:
curious why we need to associate the validator set with the verification key -- would the identifier / version number related to the circuit not be sufficient?

Reply:
This may have been unclear in the wording, trying to make it more clear now.

The version is not related to the validator set, only the VK.